hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -228,14 +251,27 @@ public RegisteredCommand register(
       datanodeDetails.setIpAddress(dnAddress.getHostAddress());
     }
     try {
+      String location;
+      if (useHostname) {
+        datanodeDetails.setNetworkName(datanodeDetails.getHostName());
+        location = nodeResolve(datanodeDetails.getHostName());
+      } else {
+        datanodeDetails.setNetworkName(datanodeDetails.getIpAddress());
+        location = nodeResolve(datanodeDetails.getIpAddress());
+      }
+      if (location != null) {
```

Comment:
Do we need to give a default location like /DEFAULT-RACK here?

Reply:
All current implemented DNSToSwitchMapping classes which resolve node to location will return "/default-rack" if can not find a match location. But this behavior is clearly defined in the interface definition. So the location null check is kept here to avoid future break of this assumption if new DNSToSwitchMapping classes are introduced.